user/serializers.py

```diff
@@ -1 +1,28 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "email", "password", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {
+            "password": {"write_only": True, "min_length": 5},
+        }
+
+    def create(self, validated_data):
+        """ Create user with encrypted password and return it. """
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        """ Update user with encrypted password and return it. """
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
```

Comment:
When updating the user, the `password` field is popped from `validated_data` and handled separately. This is correct for setting the password, but ensure that the `super().update(instance, validated_data)` call does not inadvertently reset the password to `None` if it is not provided. The current implementation should work as expected, but be cautious with any changes to the `update` logic.